### ISSUE TITLE
Hardcoded CWD in CommandUtils prevents writing migrations to AWS Lambda tmp directory.

### ISSUE THREAD

**Issue by resistdesign** [typeorm/typeorm#6731]
**Issue type:**

[ ] bug report
[x] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[ ] `mongodb`
[ ] `mssql`
[x] `mysql` / `mariadb`
[ ] `oracle`
[ ] `postgres`
[ ] `cockroachdb`
[ ] `sqlite`
[ ] `sqljs`
[ ] `react-native`
[ ] `expo`

**TypeORM version:**

[ ] `latest`
[ ] `@next`
[x] `0.x.x` (^0.2.26)

**Steps to reproduce or a small repository showing the problem:**
Try to set the migrations path to `/tmp/migrations` in an AWS Lambda function when calling `MigrationGenerateCommand.handler`.

Allowing a full path to be passed to `args.dir` would be a major benefit.

<!--
    To answer those questions you need to put "x" inside the square brackets, for example:
    [x] `mysql`
    [ ] `postgres`

    Also, please format your code properly (by taking code blocks into ```ts .... ```)
--!>

**Comment by resistdesign** [comment]
For anyone reading this, I was able to bypass the file system by creating this manager while referencing the follow files:
1. https://github.com/typeorm/typeorm/blob/4baac7108652d11af2e06a7702c18179e556adbd/src/commands/MigrationGenerateCommand.ts
1. https://github.com/typeorm/typeorm/blob/4baac7108652d11af2e06a7702c18179e556adbd/src/commands/MigrationRunCommand.ts

### WARNING: This doesn't include the "down" queries to revert a migration, but does include a transaction to make it "atomic".

```ts
import {ConnectionOptionsReader} from 'typeorm/connection/ConnectionOptionsReader';
import {Connection} from 'typeorm/connection/Connection';
import {createConnection} from 'typeorm';
import {Query} from 'typeorm/driver/Query';

/**
 * Generates a new migration file with sql needs to be executed to update schema.
 */
export class MigrationGenerateCommand {
    async handler() {
        const connectionOptionsReader = new ConnectionOptionsReader({
            root: process.cwd()
        });
        const connectionOptions = await connectionOptionsReader.get('default');
        const connection: Connection = await createConnection({
            ...connectionOptions,
            synchronize: false,
            migrationsRun: false,
            dropSchema: false,
            logging: false
        });
        const queryRunner = connection.createQueryRunner('master');
        const sqlInMemory = await connection.driver.createSchemaBuilder().log();

        if (sqlInMemory.upQueries.length) {
            await queryRunner.startTransaction();

            let failed = false;

            for (let i = 0; i < sqlInMemory.upQueries.length; i++) {
                const q: Query = sqlInMemory.upQueries[i];

                try {
                    await queryRunner.query(q.query);
                } catch (error) {
                    console.error('<<MIGRATION ERROR>>', error);
                    failed = true;
                    break;
                }
            }

            if (failed) {
                await queryRunner.rollbackTransaction();
            } else {
                await queryRunner.commitTransaction();
            }
        } else {
            console.log(`No changes in database schema were found - cannot generate a migration. To create a new empty migration use "typeorm migration:create" command`);
        }
        await connection.close();
    }
}
```

**Comment by imnotjames** [comment]
This has been fixed as of #6807

**Comment by imnotjames** [comment]
That should be out in 0.2.29 - whenever that release drops.
